### ISSUE TITLE
Simplify irqchip election

### ISSUE THREAD

**Issue by sboeuf** [cloud-hypervisor/cloud-hypervisor#511]
In the idea of simplifying the codebase, @rbradford and I discussed about removing the complex `if/else` statement trying to determine if `split_irqchip` can be used, leading to the usage of userspace `IOAPIC` or not.
The point is, if we agree that cloud-hypervisor users are running recent KVM and recent CPUs, the code logic will always find out that `TSC_DEADLINE_TIMER` is supported by the CPU, and that both `SIGNAL_MSI` and `SPLIT_IRQCHIP` capabilites are supported by KVM.

The case where the `irqchip` is fully emulated by KVM means that one IOAPIC, 2 PICs, and one PIT are emulated. There's no need for these components as a simple IOAPIC does the job.

The only concern might be about supporting Windows at some point, which might require a PIT, but even if that was the case, we could emulate the PIT in userspace anyway.

To summarize, there's not a single blocker that would prevent us from removing the full `irqchip`, leaving the VMM with the KVM split irqchip by default, and the userspace IOAPIC if needed.
